### ISSUE TITLE
Add keyboard accessibility to `Menu`

### ISSUE THREAD

**Issue by ecton** [khonsulabs/cushy#150]
We need several types of keyboard accessibility for menus:

- Typing should jump to the closest match alphabetically. Since arbitrary widgets can be used, we need a reliable way to have a type hint. This isn't meant to be a fuzzy search, just a prefix match.
- Alt shortcuts should be supported and rendered with an underline. These shortcuts should only be shown if the menu is opened via keyboard navigation or keyboard navigation is used.
- Arrow keys should be able to navigate menus and submenus.
